File: laba_9/laba_9/Fraction.cpp
#include "Fraction.h"

Fraction::Fraction(Pair p) {
	this->Setfirst(p.Getfirst());
	this->Setsecond(p.Getsecond());
}

bool Fraction::operator>(Fraction p) {
	if (this->Getfirst() > p.Getfirst() || (this->Getfirst() == p.Getfirst() && this->Getsecond() > p.Getsecond())) {
		return true;
	}
	return false;
}

bool Fraction::operator<(Fraction p) {
	if (p.Getfirst() > this->Getfirst() || (p.Getfirst() == this->Getfirst() && p.Getsecond() > this->Getsecond())) {
		return true;
	}
	return false;
}

bool Fraction::operator==(Fraction p) {
	if (p.Getfirst() == this->Getfirst() && this->Getsecond() == p.Getsecond()) {
		return true;
	}
	return false;
}

bool Fraction::operator!=(Fraction p) {
	if (p.Getfirst() == this->Getfirst() && this->Getsecond() == p.Getsecond()) {
		return false;
	}
	return true;
}

File: lab_9_C#/l_1_c/Fraction.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace l_1_c
{
    internal class Fraction : Pair
    {
		public Fraction() : base() { }
		public Fraction(int f, int s) : base(f, s) { }
		public Fraction(Pair p) : base(p) {}

		~Fraction() { }

		public static String ToString(Fraction f)
		{
			String s = "\nFirst: " + f.GetFirst() + "\nSecond: " + f.GetSecond();
			return s;
		}

		public static bool operator ==(Fraction f1, Fraction f2)
		{
			if (f1.GetFirst() == f2.GetFirst() && f1.GetSecond() == f2.GetSecond())
			{
				return true;
			}
			else
			{
				return false;
			}
		}

		public static bool operator !=(Fraction f1, Fraction f2)
		{
			if (f1.GetFirst() == f2.GetFirst() && f1.GetSecond() == f2.GetSecond()) 
			{
				return false;
			}
			else
			{
				return true;
			}
		}

		public static bool operator >(Fraction f1, Fraction f2)
		{
			if (f1.GetFirst() > f2.GetFirst() || (f1.GetFirst() == f2.GetFirst() && f1.GetSecond() > f2.GetSecond()))
			{
				return true;
			}
			else
			{
				return false;
			}
		}

		public static bool operator <(Fraction f1, Fraction f2)
		{
			if (f1.GetFirst() < f2.GetFirst() || (f1.GetFirst() == f2.GetFirst() && f1.GetSecond() < f2.GetSecond()))
			{
				return true;
			}
			else
			{
				return false;
			}
		}
	}
}
